turbine/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -76,38 +87,65 @@ fn max_coalesce_time(serialized_batch_byte_count: u64, max_batch_byte_count: u64
 }
 
 pub(super) fn recv_slot_entries(
-    receiver: &Receiver<WorkingBankEntry>,
-    carryover_entry: &mut Option<WorkingBankEntry>,
+    receiver: &Receiver<WorkingBankEntryMarker>,
+    carryover_entry: &mut Option<WorkingBankEntryMarker>,
     process_stats: &mut ProcessShredsStats,
 ) -> Result<ReceiveResults> {
     let recv_start = Instant::now();
 
     // If there is a carryover entry, use it. Else, see if there is a new entry.
-    let (mut bank, (entry, mut last_tick_height)) = match carryover_entry.take() {
-        Some((bank, (entry, tick_height))) => (bank, (entry, tick_height)),
+    let (mut bank, (entry_marker, mut last_tick_height)) = match carryover_entry.take() {
+        Some((bank, (entry_marker, tick_height))) => (bank, (entry_marker, tick_height)),
         None => receiver.recv_timeout(Duration::new(1, 0))?,
     };
     assert!(last_tick_height <= bank.max_tick_height());
-    let mut entries = vec![entry];
 
-    // Drain the channel of entries.
+    // If the first thing is a block marker, return it immediately
+    if let Some(marker) = entry_marker.as_marker().cloned() {
```

Comment:
do we need the clone here, can we add maybe an `is_marker` and then call `into_marker`?